server/chat-commands/core.ts

```diff
@@ -726,6 +729,30 @@ export const commands: ChatCommands = {
 		});
 	},
 
+	'!updatesettings': true,
+	updatesettings(target, room, user) {
+		const settings: Partial<UserSettings> = {};
+		try {
+			const raw = JSON.parse(target);
+			if (typeof raw !== 'object' || Array.isArray(raw) || !raw) {
+				this.errorReply("/updatesettings expects JSON encoded object.");
+			}
+			for (const setting of User.SETTINGS) {
```

Comment:
```suggestion
			for (const setting of Users.User.SETTINGS) {
```

Personally I would just `setting in user.settings` but if you insist on the TS-safe way...

Reply:
`user.settings` works!